src/lib/button/button.scss

```diff
@@ -7,17 +7,24 @@
 .mat-button, .mat-icon-button {
   @extend %mat-button-base;
 
-  // Only flat buttons and icon buttons (not raised or fabs) have a hover style.
-  &:hover {
-    // Use the same visual treatment for hover as for focus.
+  &[disabled]:hover {
+    &.mat-primary, &.mat-accent, &.mat-warn, .mat-button-focus-overlay {
+      background-color: transparent;
+    }
+
     .mat-button-focus-overlay {
-      opacity: 1;
+      transition: none;
+      opacity: 0;
     }
   }
+}
 
-  &[disabled]:hover {
-    &.mat-primary, &.mat-accent, &.mat-warn, .mat-button-focus-overlay {
-      background-color: transparent;
+.mat-button {
+  // Only flat buttons (not raised, fabs or icon buttons) have a hover style.
+  &:hover {
```

Comment:
Can't we do `.mat-button:hover .mat-button-focus-overlay` to avoid unnecessary nesting?